src/coord/src/timestamp.rs

```diff
@@ -333,60 +394,117 @@ fn generate_ts_updates_from_debezium(
                                     if let Value::Long(e) = v {
-                                        write_count = e;
+                                        write_count = Some(e);
                                     } else {
-                                        panic!("Incorrect AVRO format. Long expected");
+                                        bail!(
+                                            "Failed to parse Debezium transaction message. Expected long for 'event_count', got {:?}",
+                                            v
+                                        );
                                     }
                                 }
                             }
-                            if !value.is_empty() {
-                                result.push((value, write_count));
+                            match (value, write_count) {
+                                (Some(v), Some(c)) => result.push((v, c)),
+                                (v, c) => {
+                                    bail!(
+                                        "Failed to parse Debezium transaction message. Missing count or collection name. Parsed: collection={:?}, count={:?}",
+                                        v, c);
+                                }
                             }
                         } else {
-                            error!("Incorrect AVRO format. Record expected");
+                            bail!(
+                                "Failed to parse Debezium transaction message. Record expected, got {:?}",
+                                v
+                            );
                         }
                     }
+                } else {
+                    bail!(
+                        "Failed to parse Debezium transaction message. Array expected, got {:?}",
+                        value
+                    );
                 }
             } else {
-                error!(
-                    "Incorrect AVRO format. Union of Null/Array expected {:?}",
+                bail!(
+                    "Failed to parse Debezium transaction message. Union of Null/Array expected, got {:?}",
+                    value
+                );
+            }
+        } else if key == "event_count" {
+            if let Value::Union { inner: value, .. } = value {
+                if let Value::Long(count) = *value {
+                    event_count = count;
+                } else {
+                    bail!(
+                        "Failed to parse Debezium event count. Long expected, got {:?}",
+                        value
+                    );
+                }
+            } else {
+                bail!(
+                    "Failed to parse Debezium event count. Union expected, got {:?}",
                     value
                 );
             }
         }
     }
-    Some(result)
+    // TODO (chris): why can't we simply assert that event_count is non-zero?
```

Comment:
this assert checks that if the message said for example, this transaction has 9 total events, 3 from data_collection A, 4 from data_collection B and 2 from data_collection C that the sum of events from A B and C is equal to the stated total event count

asserting for non-zeroness is a different property than what is being checked here.

Reply:
Previous versions of the PR used to assert that that event_count was non-zero but our test data includes transactions with zero events (I'm pretty sure that's just bad test data). I updated this assertion to check that event_count was equivalent to the sum of all events and added this note to remind me to go back and truly figure out if event_count is always greater than zero.

Reply:
I'll open a GitHub issue for this and link against this TODO.

Reply:
Done!